### PR TITLE
Documentation for migrating from RHODS CFO to RHODS DSC

### LINKED ISSUE #7: Create documentation for migrating from RHODS CodeFlare Operator to RHODS Data Science Cluster

### PR DESCRIPTION
**Pull request by Fiona-Waters**
Closes #7 

Adding documentation with steps to migrate from RHODS CodeFlare Operator to RHODS Data Science Cluster

Verify by following the steps and ensuring a successful installation.

### CODE REVIEW

**Comment by ChristianZaccaria** [comment]
The Pre-commit is failing, you just need to add an extra empty line at the end of the file :+1:

**Review by ChristianZaccaria — CHANGES_REQUESTED** [review]
The Pre-commit is failing, you just need to add an extra empty line at the end of the file 👍

File: MIGRATION.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        kind: Subscription
+        metadata:
+        name: opendatahub-operator
+        namespace: openshift-operators
+        spec:
+        channel: fast
+        name: opendatahub-operator
+        source: community-operators
+        sourceNamespace: openshift-marketplace
+        EOF
```
Comment (ChristianZaccaria) [inline MIGRATION.md]: Nitpick, indentation is needed for it to run

```suggestion
        cat <<EOF | oc create -f -
        apiVersion: operators.coreos.com/v1alpha1
        kind: Subscription
        metadata:
          name: opendatahub-operator
          namespace: openshift-operators
        spec:
          channel: fast
          name: opendatahub-operator
          source: community-operators
          sourceNamespace: openshift-marketplace
        EOF
```

File: MIGRATION.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+## Migration Steps
+Follow these steps in order to migrate to using the updated RHODS Data Science Cluster:
+
+1. Uninstall ODH operator if you have it installed.
+    - In the [OpenShift Console](https://console.redhat.com/openshift/) navigate to Operators -> Installed Operators. Search for Open Data Hub, click on it and choose to uninstall.
+2. Uninstall RHODS CodeFlare Operator.
+     - In the [OpenShift Console](https://console.redhat.com/openshift/) navigate to Operators -> Installed Operators. Search for RHODS CodeFlare Operator, click on it and choose to uninstall.
+3. Install RHODS Version 2.4
+    - In the [OpenShift Console](https://console.redhat.com/openshift/) navigate to Operators -> OperatorHub. Search for Red Hat OpenShift Data Science and click on it. Choose version 2.4 and click Install.
```
Comment (ChristianZaccaria) [inline MIGRATION.md:14]: Currently I don't see version 2.4 in OperatorHub. I'm using QA OSD cluster. I did manage to get 2.4 via add-on though.

**Review by ChristianZaccaria — CHANGES_REQUESTED** [review]

**Review by ChristianZaccaria — COMMENTED** [review]
`kuberay-operator` in `opendatahub` namespace was leftover from old installation. It might be worth noting to remove the deployment if it's there. This might be unrelated, what do you think?

File: MIGRATION.md (hunk 1 above)
Comment (Fiona-Waters) [reply MIGRATION.md]: Good catch, thank you. Fixed this.

**Comment by Fiona-Waters** [comment]
> The Pre-commit is failing, you just need to add an extra empty line at the end of the file 👍



> The Pre-commit is failing, you just need to add an extra empty line at the end of the file 👍



> The Pre-commit is failing, you just need to add an extra empty line at the end of the file 👍

It's failing even with the newline added! seems to be also failing on some other files that I haven't touched.

**Comment by Fiona-Waters** [comment]
> `kuberay-operator` in `opendatahub` namespace was leftover from old installation. It might be worth noting to remove the deployment if it's there. This might be unrelated, what do you think?

Yes I think this should be removed also. Thanks

**Review by Bobbins228 — CHANGES_REQUESTED** [review]
Hey Fiona I just ran a `pre-commit run --all-files` on this branch and the following files were modified.
```
modified:   MIGRATION.md
modified:   config/manager/manager.yaml
modified:   config/manifests/bases/params.env
modified:   config/manifests/bases/params.yaml
modified:   config/manifests/kustomization.yaml
```

It seems you have 2 new lines at the end of MIGRATION.md which is causing issues as well as these config files

**Comment by ChristianZaccaria** [comment]
> > The Pre-commit is failing, you just need to add an extra empty line at the end of the file 👍
> 
> > The Pre-commit is failing, you just need to add an extra empty line at the end of the file 👍
> 
> > The Pre-commit is failing, you just need to add an extra empty line at the end of the file 👍
> 
> It's failing even with the newline added! seems to be also failing on some other files that I haven't touched.

Yeah, I guess there was another PR that got merged while failing pre-commit. It's now passing well done!

**Comment by ChristianZaccaria** [comment]
The generated files suggest that the version should be v1.38.0. After updating that the `verify-manifests` check should pass. I believe this could fix the `scorecard-check` too
image

**Pull request merged**